### PR TITLE
Various fixes in CanvasKit

### PR DESCRIPTION
**Pull request by hterkelsen**
- Implement `loadFontFromList` for the CanvasKit backend.
- Upgrade to CanvasKit 0.12.
- Fix crash when getting text boxes for invalid range
- Fix problem with `addOval` where the oval started from the wrong position and was drawn counterclockwise instead of clockwise
- Workaround bug when text is right-aligned or center-aligned and has layout called on it with infinite width constraints. Filed the bug here: https://bugs.chromium.org/p/skia/issues/detail?id=9874

### CODE REVIEW

File: lib/web_ui/lib/src/engine/compositor/text.dart [hunk 1]
```diff
@@ -413,10 +417,20 @@ class SkParagraph implements ui.Paragraph {
   @override
   void layout(ui.ParagraphConstraints constraints) {
     assert(constraints.width != null);
+
+    // Infinite width breaks layout, just use a very large number instead.
+    // TODO(het): Remove this once https://bugs.chromium.org/p/skia/issues/detail?id=9874
+    //            is fixed.
+    double width;
+    if (constraints.width.isInfinite) {
+      width = 1000000;
```
Comment (ferhatb) [inline lib/web_ui/lib/src/engine/compositor/text.dart]: suggestion, use: const int maxSafeInteger = 2147483647;

File: lib/web_ui/lib/src/engine/compositor/initialization.dart [hunk 2]
```diff
@@ -9,7 +9,7 @@ const bool experimentalUseSkia =
     bool.fromEnvironment('FLUTTER_WEB_USE_SKIA', defaultValue: false);
 
 /// The URL to use when downloading the CanvasKit script and associated wasm.
```
[the remark this answers is not in the document]
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/compositor/initialization.dart]: Done.

File: lib/web_ui/lib/src/engine/compositor/text.dart (hunk 1 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/compositor/text.dart]: Done.

File: lib/web_ui/lib/src/engine/compositor/fonts.dart [hunk 3]
```diff
@@ -7,8 +7,10 @@ part of engine;
 const String _robotoUrl =
```
[the remark this answers is not in the document]
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/compositor/fonts.dart]: Done.

File: lib/web_ui/lib/src/engine/compositor/initialization.dart [hunk 4]
```diff
@@ -9,7 +9,11 @@ const bool experimentalUseSkia =
...
-const String canvasKitBaseUrl = 'https://unpkg.com/canvaskit-wasm@0.11.0/bin/';
+///
+/// When CanvasKit pushes a new release to NPM, update this URL to reflect the
+/// most recent version. For example, if CanvasKit releases version 0.34.0 to
+/// NPM, update this URL to `https://unpkg.com/canvaskit-wasm@0.35.0/bin/`.
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/compositor/initialization.dart]: Why update to 35 if they release 34? :)

Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/compositor/initialization.dart]: whoops!

**Pull request merged**